test/feature/dynamic_types/DynamicTypesTests.cpp

```diff
@@ -0,0 +1,7669 @@
+
+    ASSERT_EQ(RETCODE_OK, struct_type_builder->get_member_by_name(member, "int64"));
+    ASSERT_EQ(RETCODE_OK, member->get_descriptor(md));
+    EXPECT_EQ(md->index(), 1u);
+    EXPECT_EQ(md->name(), md2->name());
+    EXPECT_EQ(md->type(), md2->type());
+
+    // • checking map indexes retrieval
+    //    + indexing by id
+    DynamicTypeMembersById members_by_id;
+    struct_type_builder->get_all_members(members_by_id);
+    EXPECT_EQ(2, members_by_id.size());
+
+    auto dm3 = members_by_id[3];
+    ASSERT_TRUE(dm3);
+    ASSERT_EQ(RETCODE_OK, dm3->get_descriptor(md));
+    EXPECT_EQ(md->index(), 0u);
+    EXPECT_EQ(md->name(), md1->name());
+    EXPECT_EQ(md->type(), md1->type());
+
+    auto dm1 = members_by_id[1];
+    ASSERT_TRUE(dm1);
+    ASSERT_EQ(RETCODE_OK, dm1->get_descriptor(md));
+    EXPECT_EQ(md->index(), 1u);
+    EXPECT_EQ(md->name(), md2->name());
+    EXPECT_EQ(md->type(), md2->type());
+
+    //    + indexing by name
+    DynamicTypeMembersByName members_by_name;
+    struct_type_builder->get_all_members_by_name(members_by_name);
+    EXPECT_EQ(2, members_by_name.size());
+
+    dm3 = members_by_name["int32"];
+    ASSERT_EQ(RETCODE_OK, dm3->get_descriptor(md));
+    EXPECT_EQ(md->index(), 0u);
+    EXPECT_EQ(md->name(), md1->name());
+    EXPECT_EQ(md->type(), md1->type());
+
+    dm1 = members_by_name["int64"];
+    ASSERT_EQ(RETCODE_OK, dm1->get_descriptor(md));
+    EXPECT_EQ(md->index(), 1u);
+    EXPECT_EQ(md->name(), md2->name());
+    EXPECT_EQ(md->type(), md2->type());
+
+    // • checking indexes work according with OMG standard 1.3 section 7.5.2.7.6
+    md = traits<MemberDescriptor>::make_shared();
+    md->id(7);
+    md->name("bool");
+    md->type(factory->get_primitive_type(eprosima::fastdds::dds::TK_BOOLEAN));
+    ASSERT_EQ(RETCODE_OK, struct_type_builder->add_member(md));
+
+    struct_type_builder->get_all_members(members_by_id);
+    ASSERT_EQ(3, members_by_id.size());
+
+    MemberDescriptor::_ref_type tmp = traits<MemberDescriptor>::make_shared();
+    auto dm = members_by_id[3];
+    ASSERT_EQ(RETCODE_OK, dm->get_descriptor(tmp));
+    EXPECT_EQ(tmp->index(), 0u);
+    EXPECT_EQ(tmp->name(), md1->name());
+    EXPECT_EQ(tmp->type(), md1->type());
+
+    dm = members_by_id[7];
+    ASSERT_EQ(RETCODE_OK, dm->get_descriptor(tmp));
+    EXPECT_EQ(tmp->name(), md->name());
+    EXPECT_EQ(tmp->type(), md->type());
+
+    dm = members_by_id[1];
+    ASSERT_EQ(RETCODE_OK, dm->get_descriptor(tmp));
+    EXPECT_EQ(tmp->index(), 1u);
+    EXPECT_EQ(tmp->name(), md2->name());
+    EXPECT_EQ(tmp->type(), md2->type());
+
+
+    // • checking adding duplicates
+    {
+        eprosima::fastdds::dds::Log::ScopeLogs _("disable");
+        //    + duplicate name
+        md = traits<MemberDescriptor>::make_shared();
+        md->id(1);
+        md->name("int32");
+        md->type(factory->get_primitive_type(eprosima::fastdds::dds::TK_INT32));
+        EXPECT_NE(RETCODE_OK, struct_type_builder->add_member(md));
```

Comment:
More explicit and clear. I suggest applying this to any other similar occurrence.

```suggestion
        EXPECT_EQ(RETCODE_BAD_PARAMETER, struct_type_builder->add_member(md));
```

Reply:
Done in 2a37cf615